src/server/pps/server/api_server.go

```diff
@@ -464,6 +467,248 @@ func (a *apiServer) RestartDatum(ctx context.Context, request *pps.RestartDatumR
+
+	return &pps.DatumInfos{datumInfos}, nil
+}
+
+type byDatumStateThenTime []*pps.DatumInfo
+
+func (a byDatumStateThenTime) Len() int      { return len(a) }
+func (a byDatumStateThenTime) Swap(i, j int) { a[i], a[j] = a[j], a[i] }
+func (a byDatumStateThenTime) Less(i, j int) bool {
+	byState := a[i].State < a[j].State
+	if a[i].State != a[j].State {
+		return byState
+	}
+	if a[i].Stats == nil || a[j].Stats == nil {
+		return byState
+	}
+	return client.GetDatumTotalTime(a[i].Stats) > client.GetDatumTotalTime(a[j].Stats)
+}
+
+func (a *apiServer) getDatum(ctx context.Context, repo string, commit *pfs.Commit, jobID string, datumID string) (*pps.DatumInfo, error) {
+	pfsClient, err := a.getPFSClient()
+	if err != nil {
+		return nil, err
+	}
+	// Populate status
+	state := pps.DatumState_FAILED
+	stateFile := &pfs.File{
+		Commit: commit,
+		Path:   fmt.Sprintf("/%v/%v/failure", jobID, datumID),
+	}
+	getFileClient, err := pfsClient.GetFile(ctx, &pfs.GetFileRequest{stateFile, 0, 0})
+	if err := grpcutil.WriteFromStreamingBytesClient(getFileClient, ioutil.Discard); err != nil {
+		if strings.Contains(err.Error(), "not found") {
+			state = pps.DatumState_SUCCESS
+		} else {
+			return nil, err
+		}
+	}
+
+	// Populate stats
+	statsFile := &pfs.File{
+		Commit: commit,
+		Path:   fmt.Sprintf("/%v/%v/stats", jobID, datumID),
+	}
+	getFileClient, err = pfsClient.GetFile(ctx, &pfs.GetFileRequest{statsFile, 0, 0})
+	if err != nil {
+		return nil, err
+	}
+	r, w := io.Pipe()
+	var eg errgroup.Group
+	eg.Go(func() error {
+		if err := grpcutil.WriteFromStreamingBytesClient(getFileClient, w); err != nil {
+			return err
+		}
+		w.Close()
+		return nil
+	})
+	stats := &pps.ProcessStats{}
+	err = jsonpb.Unmarshal(r, stats)
+	if err != nil {
+		return nil, err
+	}
+	if err = eg.Wait(); err != nil {
+		return nil, err
+	}
+
+	return &pps.DatumInfo{
+		Datum: &pps.Datum{datumID},
+		State: state,
+		Stats: stats,
+		PfsState: &pfs.File{
+			Commit: commit,
+			Path:   fmt.Sprintf("/%v/%v/pfs", jobID, datumID),
+		},
+	}, nil
+}
+
+func (a *apiServer) InspectDatum(ctx context.Context, request *pps.InspectDatumRequest) (response *pps.DatumInfo, retErr error) {
+	func() { a.Log(request, nil, nil, 0) }()
+	defer func(start time.Time) { a.Log(request, response, retErr, time.Since(start)) }(time.Now())
+	jobInfo, err := a.InspectJob(ctx, &pps.InspectJobRequest{
+		Job: &pps.Job{
+			ID: request.Job.ID,
+		},
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	pfsClient, err := a.getPFSClient()
+	if err != nil {
+		return
+	}
+	// Check 'stats' branch exists
+	branches, err := pfsClient.ListBranch(ctx, &pfs.ListBranchRequest{jobInfo.OutputCommit.Repo})
+	if err != nil {
+		return nil, err
+	}
+	var statsBranch *pfs.BranchInfo
+	for _, branch := range branches.BranchInfo {
+		if branch.Name == "stats" {
+			statsBranch = branch
+		}
+	}
+	if statsBranch == nil {
+		return nil, fmt.Errorf("stats branch not found on %v", jobInfo.OutputCommit.Repo.Name)
+	}
```

Comment:
Again, no need to look for the stats branch, `jobInfo` has the correct commit to use in it.